### ISSUE TITLE
Why are losses during training so low and results so bad, what is causing this behavior?

### ISSUE THREAD

**Issue by cepa995** [facebookresearch/detectron2#2380]
I am trying to train detectron2 for detecting 1 class only. I've created and registered custom dataset and inspected to see if bboxes are correct. When I use default trainer (as it is in tutorial):

```
cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/faster_rcnn_R_101_C4_3x.yaml"))
cfg.DATASETS.TRAIN = ("breast_mammography_train",)
cfg.DATASETS.TEST = ()   # no metrics implemented for this dataset
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/faster_rcnn_R_101_C4_3x.yaml")  # Let training initialize from model zoo
cfg.DATALOADER.NUM_WORKERS = 2
cfg.SOLVER.IMS_PER_BATCH = 2
cfg.SOLVER.BASE_LR = 0.00025
cfg.SOLVER.MAX_ITER = 500   
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 512   # faster, and good enough for this toy dataset
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg)
trainer.resume_or_load(resume=False)
trainer.train()
```
These are results I am getting:
```
[12/11 20:48:48 d2.checkpoint.c2_model_loading]: Some model parameters or buffers are not found in the checkpoint:
  pixel_mean
  pixel_std
  proposal_generator.anchor_generator.cell_anchors.0
  proposal_generator.rpn_head.anchor_deltas.{bias, weight}
  proposal_generator.rpn_head.conv.{bias, weight}
  proposal_generator.rpn_head.objectness_logits.{bias, weight}
  roi_heads.box_predictor.bbox_pred.{bias, weight}
  roi_heads.box_predictor.cls_score.{bias, weight}
[12/11 20:48:48 d2.checkpoint.c2_model_loading]: The checkpoint state_dict contains keys that are not used by the model:
  fc1000_b
  fc1000_w
[12/11 20:48:49 d2.engine.train_loop]: Starting training from iteration 0
[12/11 20:49:25 d2.utils.events]:  eta: 0:13:58  iter: 19  total_loss: 1.496  loss_cls: 0.7705  loss_box_reg: 0.0009855  loss_rpn_cls: 0.7079  loss_rpn_loc: 0.01184  time: 1.7501  data_time: 0.1114  lr: 4.9953e-06  max_mem: 11756M
[12/11 20:50:00 d2.utils.events]:  eta: 0:13:31  iter: 39  total_loss: 1.01  loss_cls: 0.2988  loss_box_reg: 0.001118  loss_rpn_cls: 0.6829  loss_rpn_loc: 0.009809  time: 1.7608  data_time: 0.0318  lr: 9.9902e-06  max_mem: 11756M
[12/11 20:50:37 d2.utils.events]:  eta: 0:13:06  iter: 59  total_loss: 0.7532  loss_cls: 0.09395  loss_box_reg: 0.0007173  loss_rpn_cls: 0.6347  loss_rpn_loc: 0.02043  time: 1.7837  data_time: 0.0079  lr: 1.4985e-05  max_mem: 11756M
[12/11 20:51:14 d2.utils.events]:  eta: 0:12:33  iter: 79  total_loss: 0.6255  loss_cls: 0.05622  loss_box_reg: 0.0005691  loss_rpn_cls: 0.563  loss_rpn_loc: 0.01026  time: 1.7986  data_time: 0.0238  lr: 1.998e-05  max_mem: 11756M
[12/11 20:51:51 d2.utils.events]:  eta: 0:12:05  iter: 99  total_loss: 0.5472  loss_cls: 0.04413  loss_box_reg: 0.000922  loss_rpn_cls: 0.4789  loss_rpn_loc: 0.02123  time: 1.8165  data_time: 0.0107  lr: 2.4975e-05  max_mem: 11756M
[12/11 20:52:29 d2.utils.events]:  eta: 0:11:34  iter: 119  total_loss: 0.4381  loss_cls: 0.03882  loss_box_reg: 0.0005709  loss_rpn_cls: 0.3881  loss_rpn_loc: 0.01036  time: 1.8249  data_time: 0.0146  lr: 2.997e-05  max_mem: 11756M
[12/11 20:53:07 d2.utils.events]:  eta: 0:11:02  iter: 139  total_loss: 0.3501  loss_cls: 0.0363  loss_box_reg: 0.0004903  loss_rpn_cls: 0.2961  loss_rpn_loc: 0.009207  time: 1.8400  data_time: 0.0334  lr: 3.4965e-05  max_mem: 11756M
[12/11 20:53:46 d2.utils.events]:  eta: 0:10:32  iter: 159  total_loss: 0.2931  loss_cls: 0.0349  loss_box_reg: 0.002525  loss_rpn_cls: 0.2365  loss_rpn_loc: 0.01573  time: 1.8510  data_time: 0.0076  lr: 3.996e-05  max_mem: 11756M
[12/11 20:54:25 d2.utils.events]:  eta: 0:09:56  iter: 179  total_loss: 0.2207  loss_cls: 0.03532  loss_box_reg: 0.001892  loss_rpn_cls: 0.1792  loss_rpn_loc: 0.01044  time: 1.8599  data_time: 0.0048  lr: 4.4955e-05  max_mem: 11756M
[12/11 20:55:03 d2.utils.events]:  eta: 0:09:22  iter: 199  total_loss: 0.2016  loss_cls: 0.02987  loss_box_reg: 0.002721  loss_rpn_cls: 0.1456  loss_rpn_loc: 0.0127  time: 1.8667  data_time: 0.0096  lr: 4.995e-05  max_mem: 11756M
[12/11 20:55:43 d2.utils.events]:  eta: 0:08:47  iter: 219  total_loss: 0.1476  loss_cls: 0.02903  loss_box_reg: 0.003859  loss_rpn_cls: 0.1076  loss_rpn_loc: 0.009942  time: 1.8760  data_time: 0.0051  lr: 5.4945e-05  max_mem: 11756M
[12/11 20:56:21 d2.utils.events]:  eta: 0:08:11  iter: 239  total_loss: 0.1706  loss_cls: 0.03459  loss_box_reg: 0.01034  loss_rpn_cls: 0.09521  loss_rpn_loc: 0.01317  time: 1.8816  data_time: 0.0046  lr: 5.994e-05  max_mem: 11756M
[12/11 20:57:01 d2.utils.events]:  eta: 0:07:35  iter: 259  total_loss: 0.1174  loss_cls: 0.0308  loss_box_reg: 0.005168  loss_rpn_cls: 0.07858  loss_rpn_loc: 0.01011  time: 1.8897  data_time: 0.0047  lr: 6.4935e-05  max_mem: 11757M
[12/11 20:57:41 d2.utils.events]:  eta: 0:06:58  iter: 279  total_loss: 0.1357  loss_cls: 0.03018  loss_box_reg: 0.005391  loss_rpn_cls: 0.07308  loss_rpn_loc: 0.01102  time: 1.8959  data_time: 0.0064  lr: 6.993e-05  max_mem: 11757M
[12/11 20:58:20 d2.utils.events]:  eta: 0:06:22  iter: 299  total_loss: 0.1052  loss_cls: 0.02538  loss_box_reg: 0.004834  loss_rpn_cls: 0.06466  loss_rpn_loc: 0.01031  time: 1.9004  data_time: 0.0051  lr: 7.4925e-05  max_mem: 11757M
[12/11 20:58:59 d2.utils.events]:  eta: 0:05:45  iter: 319  total_loss: 0.09958  loss_cls: 0.02411  loss_box_reg: 0.009482  loss_rpn_cls: 0.05033  loss_rpn_loc: 0.007759  time: 1.9026  data_time: 0.0072  lr: 7.992e-05  max_mem: 11757M
[12/11 20:59:37 d2.utils.events]:  eta: 0:05:06  iter: 339  total_loss: 0.08288  loss_cls: 0.0193  loss_box_reg: 0.004392  loss_rpn_cls: 0.05143  loss_rpn_loc: 0.008958  time: 1.9043  data_time: 0.0067  lr: 8.4915e-05  max_mem: 11757M
[12/11 21:00:17 d2.utils.events]:  eta: 0:04:28  iter: 359  total_loss: 0.08301  loss_cls: 0.02419  loss_box_reg: 0.006458  loss_rpn_cls: 0.04459  loss_rpn_loc: 0.008127  time: 1.9076  data_time: 0.0239  lr: 8.991e-05  max_mem: 11757M
[12/11 21:00:56 d2.utils.events]:  eta: 0:03:50  iter: 379  total_loss: 0.1249  loss_cls: 0.03129  loss_box_reg: 0.01382  loss_rpn_cls: 0.05255  loss_rpn_loc: 0.01186  time: 1.9094  data_time: 0.0052  lr: 9.4905e-05  max_mem: 11757M
[12/11 21:01:35 d2.utils.events]:  eta: 0:03:12  iter: 399  total_loss: 0.09116  loss_cls: 0.02433  loss_box_reg: 0.007517  loss_rpn_cls: 0.04316  loss_rpn_loc: 0.009138  time: 1.9114  data_time: 0.0096  lr: 9.99e-05  max_mem: 11757M
[12/11 21:02:14 d2.utils.events]:  eta: 0:02:34  iter: 419  total_loss: 0.08744  loss_cls: 0.02771  loss_box_reg: 0.01507  loss_rpn_cls: 0.03911  loss_rpn_loc: 0.006996  time: 1.9136  data_time: 0.0050  lr: 0.0001049  max_mem: 11757M
[12/11 21:02:53 d2.utils.events]:  eta: 0:01:55  iter: 439  total_loss: 0.1013  loss_cls: 0.0253  loss_box_reg: 0.009539  loss_rpn_cls: 0.04704  loss_rpn_loc: 0.0108  time: 1.9154  data_time: 0.0113  lr: 0.00010989  max_mem: 11757M
[12/11 21:03:32 d2.utils.events]:  eta: 0:01:17  iter: 459  total_loss: 0.126  loss_cls: 0.03074  loss_box_reg: 0.0159  loss_rpn_cls: 0.05007  loss_rpn_loc: 0.01207  time: 1.9164  data_time: 0.0068  lr: 0.00011489  max_mem: 11757M
[12/11 21:04:11 d2.utils.events]:  eta: 0:00:38  iter: 479  total_loss: 0.1028  loss_cls: 0.03973  loss_box_reg: 0.01773  loss_rpn_cls: 0.04252  loss_rpn_loc: 0.01075  time: 1.9174  data_time: 0.0104  lr: 0.00011988  max_mem: 11757M
[12/11 21:04:51 d2.utils.events]:  eta: 0:00:00  iter: 499  total_loss: 0.15  loss_cls: 0.04504  loss_box_reg: 0.02843  loss_rpn_cls: 0.04517  loss_rpn_loc: 0.00853  time: 1.9188  data_time: 0.0043  lr: 0.00012488  max_mem: 11757M
[12/11 21:04:52 d2.engine.hooks]: Overall training speed: 498 iterations in 0:15:55 (1.9188 s / it)
[12/11 21:04:52 d2.engine.hooks]: Total training time: 0:15:58 (0:00:02 on hooks)
```

On the other hand, if I choose `COCO-Detection/retinanet_R_50_FPN_3x.yaml` as my model, training is some what normal:

```
[12/11 21:06:09 d2.checkpoint.c2_model_loading]: Some model parameters or buffers are not found in the checkpoint:
  anchor_generator.cell_anchors.{0, 1, 2, 3, 4}
  backbone.fpn_lateral3.{bias, weight}
  backbone.fpn_lateral4.{bias, weight}
  backbone.fpn_lateral5.{bias, weight}
  backbone.fpn_output3.{bias, weight}
  backbone.fpn_output4.{bias, weight}
  backbone.fpn_output5.{bias, weight}
  backbone.top_block.p6.{bias, weight}
  backbone.top_block.p7.{bias, weight}
  head.bbox_pred.{bias, weight}
  head.bbox_subnet.0.{bias, weight}
  head.bbox_subnet.2.{bias, weight}
  head.bbox_subnet.4.{bias, weight}
  head.bbox_subnet.6.{bias, weight}
  head.cls_score.{bias, weight}
  head.cls_subnet.0.{bias, weight}
  head.cls_subnet.2.{bias, weight}
  head.cls_subnet.4.{bias, weight}
  head.cls_subnet.6.{bias, weight}
  pixel_mean
  pixel_std
[12/11 21:06:09 d2.checkpoint.c2_model_loading]: The checkpoint state_dict contains keys that are not used by the model:
  fc1000_b
  fc1000_w
  conv1_b
[12/11 21:06:09 d2.engine.train_loop]: Starting training from iteration 0
[12/11 21:06:33 d2.utils.events]:  eta: 0:08:56  iter: 19  total_loss: 1.748  loss_cls: 1.176  loss_box_reg: 0.5771  time: 1.1974  data_time: 0.8637  lr: 4.9953e-06  max_mem: 11757M
[12/11 21:06:56 d2.utils.events]:  eta: 0:08:27  iter: 39  total_loss: 2.082  loss_cls: 1.43  loss_box_reg: 0.6613  time: 1.1638  data_time: 0.7881  lr: 9.9902e-06  max_mem: 11757M
[12/11 21:07:18 d2.utils.events]:  eta: 0:07:43  iter: 59  total_loss: 2.111  loss_cls: 1.389  loss_box_reg: 0.7036  time: 1.1462  data_time: 0.7650  lr: 1.4985e-05  max_mem: 11757M
[12/11 21:07:40 d2.utils.events]:  eta: 0:07:13  iter: 79  total_loss: 2.279  loss_cls: 1.541  loss_box_reg: 0.7378  time: 1.1312  data_time: 0.7391  lr: 1.998e-05  max_mem: 11757M
```

I am not looking for a "how to train my model" just a pointer in the right direction if I am doing something wrong or why is the above behavior produced? I've expected (e.g. bounding box regression) loss to be in normal ranges. I do not understand why does loss fluctuate so much and what does it mean if its so low. 

Also, if you look at code above it prints that "_The checkpoint state_dict contains keys that are not used by the model_" is this to be expected?

Typically low loss means good results (even if it is overfitting), but my results are horrible. Any kind of information would be appreciated. MUCH THANKS!

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.

>  if you look at code above it prints that "The checkpoint state_dict contains keys that are not used by the model" is this to be expected?

No, and the tutorial does not print this. I think your code is probably different from what is shown above. If you need help to solve an unexpected issue you observed, please include details following the ["Unexpected behaviors" issue template](https://github.com/facebookresearch/detectron2/issues/new/choose).
